cmd/clientd/internal/mechanism/demand/demand.go

```diff
@@ -0,0 +1,121 @@
+// Package demand defines the on-demand syncing mechanism, syncing on-demand
+// via an HTTP request.
+package demand
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/george-e-shaw-iv/outflux/internal/log"
+)
+
+// Syncer implements the mechanism.Syncer type for the on-demand mechanism.
+type Syncer struct {
+	cfg *Config
+
+	sync   chan struct{}
+	server *http.Server
+
+	once sync.Once
+}
+
+// NewSyncer validates the configuration passed in and returns a new syncer
+// for interval on-demand syncing.
+func NewSyncer(cfg *Config) (*Syncer, error) {
+	if err := cfg.Validate(); err != nil {
+		return nil, fmt.Errorf("validate on-demand config: %w", err)
+	}
+
+	return &Syncer{
+		cfg: cfg,
+	}, nil
+}
+
+// handler is the HTTP handler function used to intake the on-demand request
+// to sync.
+func (s *Syncer) handler(w http.ResponseWriter, r *http.Request) {
+	if len(s.sync) == 0 {
+		// Only send a sync signal if there isn't already one waiting.
+		s.sync <- struct{}{}
+	}
+
+	w.WriteHeader(http.StatusNoContent)
+}
+
+// initialize initializes an HTTP server and channel to perform on-demand
+// syncing via HTTP.
+func (s *Syncer) initialize() {
+	endpoint := strings.TrimSuffix(strings.TrimPrefix(s.cfg.Endpoint, "/"), "/")
+
+	mux := http.NewServeMux()
+	mux.HandleFunc(endpoint, s.handler)
+
+	s.sync = make(chan struct{}, 1)
+	s.server = &http.Server{
+		Addr:           fmt.Sprintf(":%d", s.cfg.Port),
```

Comment:
Do we want to be able to set the bind address as well? e.g. `0.0.0.0` or `127.0.0.1`?

Reply:
Might as well make the host configurable and let the user decide, good call